plugins/producer_plugin/test/test_read_only_trx.cpp

```diff
@@ -0,0 +1,147 @@
+#define BOOST_TEST_MODULE producer_read_only_trxs
+#include <boost/test/included/unit_test.hpp>
+
+#include <eosio/producer_plugin/producer_plugin.hpp>
+#include <eosio/testing/tester.hpp>
+#include <eosio/chain/genesis_state.hpp>
+#include <eosio/chain/thread_utils.hpp>
+#include <eosio/chain/transaction_metadata.hpp>
+#include <eosio/chain/trace.hpp>
+#include <eosio/chain/name.hpp>
+#include <eosio/chain/application.hpp>
+
+namespace eosio::test::detail {
+using namespace eosio::chain::literals;
+struct testit {
+   uint64_t      id;
+   testit( uint64_t id = 0 )
+         :id(id){}
+   static account_name get_account() {
+      return chain::config::system_account_name;
+   }
+   static action_name get_name() {
+      return "testit"_n;
+   }
+};
+}
+FC_REFLECT( eosio::test::detail::testit, (id) )
+
+namespace {
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::test::detail;
+
+auto make_unique_trx( const chain_id_type& chain_id ) {
+   static uint64_t nextid = 0;
+   ++nextid;
+   account_name creator = config::system_account_name;
+   signed_transaction trx;
+   trx.actions.emplace_back( vector<permission_level>{{creator, config::active_name}}, testit{nextid} );
```

Comment:
Every read-only trx is failing with:
`auth: eosio, action: eosio:testit, read-only action 'testit' cannot have authorizations`

Reply:
Thanks. Will do.